java-squid/src/main/java/org/sonar/java/model/VisitorsBridge.java

```diff
@@ -116,14 +120,16 @@ public void visitFile(@Nullable AstNode astNode) {
       } else {
         SemanticModel.handleMissingTypes(tree);
       }
-      JavaFileScannerContext context = new DefaultJavaFileScannerContext(tree, (SourceFile) getContext().peekSourceCode(), getContext().getFile(), semanticModel, analyseAccessors);
-      for (JavaFileScanner scanner : scanners) {
+    }
+    JavaFileScannerContext context = new DefaultJavaFileScannerContext(tree, (SourceFile) getContext().peekSourceCode(), getContext().getFile(), semanticModel, analyseAccessors);
+    for (JavaFileScanner scanner : scanners) {
+      if(astNode != null || scanner instanceof AstScannerExceptionHandler) {
```

Comment:
This is a bit crappy but works, this is to avoid to actually call a scan on null astNode except for AstScannerExceptionHandler's